addons/repair/models/repair.py

```diff
@@ -101,11 +101,23 @@ def _default_picking_type_id(self):
         'procurement.group', 'Procurement Group',
         copy=False)
     location_id = fields.Many2one(
-        'stock.location', 'Location',
+        'stock.location', 'Components Location',
         compute="_compute_location_id",
         store=True, readonly=False, required=True, precompute=True,
         index=True, check_company=True,
+        help="This is the location where the components of product to repair is located.")
+    default_product_location_src_id = fields.Many2one(
+        'stock.location', 'Product Source Location',
+        compute="_compute_default_product_location_src_id",
+        store=True, readonly=False, required=True, precompute=True,
+        index=True, check_company=True,
         help="This is the location where the product to repair is located.")
+    default_product_location_dest_id = fields.Many2one(
+        'stock.location', 'Product Destination Location',
+        compute="_compute_default_product_location_dest_id",
+        store=True, readonly=False, required=True, precompute=True,
+        index=True, check_company=True,
+        help="This is the location where the repaired product is located.")
```

Comment:
- Don't make new field you can make it related
- don't define useless attribute if it's not required